cryptoconditions/types/ed25519.py

```diff
@@ -137,22 +166,48 @@ def parse_dict(self, data):
         Returns:
             Fulfillment
         """
-        self.public_key = VerifyingKey(data['public_key'])
-        self.signature = base58.b58decode(data['signature']) if data['signature'] else None
+        self.public_key = base58.b58decode(data['public_key'])
+        if data['signature']:
+            self.signature = base58.b58decode(data['signature'])
+
+    # TODO Adapt according to outcomes of
+    # https://github.com/rfcs/crypto-conditions/issues/16
+    def parse_json(self, data):
+        """
+        Generate fulfillment payload from a dict
+
+        Args:
+            data (dict): description of the fulfillment
+
+        Returns:
+            Fulfillment
+        """
+        self.public_key = urlsafe_b64decode(base64_add_padding(
+            data['publicKey']))
+        self.signature = urlsafe_b64decode(base64_add_padding(
+            data['signature']))
+
+    def parse_asn1_dict_payload(self, data):
+        self.public_key = data['publicKey']
+        self.signature = data['signature']
 
     def validate(self, message=None, **kwargs):
```

Comment:
I don't think there is a reason for `message` to be an optional argument. `verify` should fail if message is `None`. Also I don't see a reason for `**kwargs`